File: PythonRandomClient/ai.py
# -*- coding: utf-8 -*-

# python imports
import random

# chillin imports
from chillin_client import RealtimeAI

# project imports
from ks.models import World, EDirection
from ks.commands import ChangeGhostDirection, ChangePacmanDirection


class AI(RealtimeAI):

    def __init__(self, world):
        super(AI, self).__init__(world)


    def initialize(self):
        print('initialize')


    def decide(self):
        print('decide')

        if self.my_side == 'Pacman':
            self.send_command(ChangePacmanDirection(random.choice(list(EDirection))))
        elif self.my_side == 'Ghost':
            for ghost in self.world.ghosts:
                self.send_command(ChangeGhostDirection(ghost.id, random.choice(list(EDirection))))

File: CSharpClient/Game/AI.cs
using System;

using KoalaTeam.Chillin.Client;
using KS;
using KS.Commands;
using KS.Models;

using KSObject = KS.KSObject;

namespace Game
{
	public class AI : RealtimeAI<World, KSObject>
	{
		private readonly Random random = new Random();

		public AI(World world) : base(world)
		{
		}

		public override void Initialize()
		{
			Console.WriteLine("initialize");
		}

		public override void Decide()
		{
			Console.WriteLine("decide");

			if (this.MySide == "Pacman")
			{
				ChangePacmanDirection((EDirection)random.Next(Enum.GetNames(typeof(EDirection)).Length));
			}
			else if (this.MySide == "Ghost")
			{
				foreach (var ghost in this.World.Ghosts)
					ChangeGhostDirection(
						ghost.Id,
						(EDirection)random.Next(Enum.GetNames(typeof(EDirection)).Length)
					);
			}
		}


		public void ChangePacmanDirection(EDirection direction)
		{
			this.SendCommand(new ChangePacmanDirection() { Direction = (ECommandDirection?)direction });
		}

		public void ChangeGhostDirection(int? id, EDirection direction)
		{
			this.SendCommand(new ChangeGhostDirection() { Id = id, Direction = (ECommandDirection?)direction });
		}
	}
}
